Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -1,17 +1,31 @@
 function Add-Certificate {
     [CmdletBinding()]
-    param([Parameter(Mandatory=$true)]$Endpoint)
+    param(
+        [Parameter(Mandatory=$true)] $Endpoint,
+        [Switch] $ServicePrincipal
+    )
 
     # Add the certificate to the cert store.
-    $bytes = [System.Convert]::FromBase64String($Endpoint.Auth.Parameters.Certificate)
     $certificate = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
-    $certificate.Import($bytes)
+
+    if ($ServicePrincipal) {
+        $pemFileContent = $Endpoint.Auth.Parameters.ServicePrincipalCertificate
+        $pfxFilePath, $pfxFilePassword = ConvertTo-Pfx -pemFileContent $pemFileContent
+        
+        $certificate.Import($pfxFilePath, $pfxFilePassword, [System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::PersistKeySet)
```

Comment:
Who removes from the store after execution?

Reply:
Added logic to remove certificate from store once task execution completes.